DigitalLearningSolutions.Web/Views/RegisterAtNewCentre/Confirmation.cshtml

```diff
@@ -21,12 +22,25 @@
       <p>Your delegate number is <span class="nhsuk-u-font-weight-bold">@Model.CandidateNumber</span></p>
     </div>
 
+    <vc:verify-email-warning-if-appropriate
+      is-choose-a-centre-page="false"
+      is-my-account-page="false"
+      is-registration-journey="true"
+      unverified-primary-email="@Model.UnverifiedPrimaryEmail"
+      unverified-centre-emails="@Model.UnverifiedCentreEmails" />
+
+    @if (emailsAreVerified) {
+      <vc:action-link asp-controller="Login" asp-action="ChooseACentre" link-text="Switch centre" />
+    } else {
+      <vc:action-link asp-controller="MyAccount" asp-action="Index" link-text="View My account" />
+    }
+
     @if (Model.HasAdminAccountAtCentre) {
       <p>
         You will need to log out and in again before you can access the Learning Portal.
       </p>
       <partial name="_LogoutForm" model="@ButtonType.Secondary" />
-    } else if (Model.Approved) {
+    } else if (Model.Approved && emailsAreVerified) {

```

Comment:
I've cleaned this up a bit. The only thing I'm not certain of is the priority of the action links/buttons. I've put them in this order but feel free to disagree:
1. Logout button (For when the user has an admin account at that centre. They should be auto-approved and they might still have an unverified email, but the verify email warning tells them to visit the My Account page if they need to resend the verification link, so prompting them to log out seems more important)
2. Switch centre button (For when the account is approved and the email is verified)
3. Centre contact info action link (For when the account if not approved. The email might still be unverified but the verify email warning tells them to visit the My Account page if they need to resend the verification link, so showing them where to find centre contact info seems more important as they might not otherwise know where to find it.)
4. View My Account action link (For if the email is unverified but all else is well - account approved, no admin account at the centre.)

Reply:
So actually, I think we should have the switch centre button for the admin as well. I don't think we need to treat them separately from other auto-approved users. All they need to do is re-log-into the current centre. If we log them out then they have to enter credentials again which'll be a tad annoying for them. Can we just show them the standard auto-approved text?

It feels strange to send them to MyAccount if the email is unverified - they'll still be logged into whichever centre they're currently logged into and so they can't even see the centre email they need to verify. I think switch-centre is probably still suitable for them - it's where they'll want to go after verifying their email anyway.

Reply:
Sure, I've merged the HasAdminAccount condition with the one that shows the switch centre button